src/client/driver/driver.js

```diff
@@ -248,17 +254,52 @@ export default class Driver extends serviceUtils.EventEmitter {
         const childWindowDriverLink = new ChildWindowDriverLink(e.window, e.windowId);
 
         this.childWindowDriverLinks.push(childWindowDriverLink);
+        this._ensureChildWindowClosedWatcher();
+    }
+
+    _ensureChildWindowClosedWatcher () {
+        if (this.checkChildWindowClosedIntervalId)
+            return;
+
+        this.checkChildWindowClosedIntervalId = nativeMethods.setInterval.call(window, () => {
+            for (const childWindowDriverLink of this.childWindowDriverLinks) {
+                if (childWindowDriverLink.driverWindow.closed) {
+                    arrayUtils.remove(this.childWindowDriverLinks, childWindowDriverLink);
+
+                    if (this.role === DriverRole.replica)
+                        this._setCurrentWindowAsMaster();
```

Comment:
This code supposes that we can have multiple `childWindowDriverLinks` and I suppose that in some cases we call `this._setCurrentWindowAsMaster` multiple times. Or not?

Reply:
Yes, you are right. I will fix it.